src/components/forms/forms.js

```diff
@@ -13,80 +21,122 @@ import {
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
   CHECKBOX: {
     defaultValue: (config) => (config?.valueType === 'string' ? '' : false),
-    component: formInputCheckbox
+    component: formInputCheckbox,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+    ],
   },
   SELECT: {
     defaultValue: (config) => (config?.isMulti ? [] : ''),
-    updateDefaultValue: ({ selectOptions, defaultValue }) => {
-      const isStringArr = isString(selectOptions[0]);
-      return isStringArr
-        ? { key: defaultValue, displayValue: defaultValue }
-        : defaultValue;
-    },
-    component: formInputSelect
+    updateDefaultValue: updateSelectDefaultValue,
+    component: formInputSelect,
+    requiredProps: [
+      'selectOptions'
+      // 'selectOptions' example:
+      // [
+      //  {displayText: 'Option 1'}
+      //   ^^^ can pass other fields in as extra info, e.g.:
+      //  {displayText: 'Dac 1', dacId: 213}
+      // ]
+    ],
+    optionalProps: [
+      'isCreatable', // allows user to input their own
+      'isMulti',
+      'placeholder',
+      'ariaDescribedby',
+      'required',
+      'selectConfig',
+    ],
+    customPropValidation: customSelectPropValidation,
   },
 };
 
 export const FormValidators = {
-  REQUIRED: {
-    isValid: (value) => value !== undefined && value !== null && value !== '',
-    msg: 'Please enter a value'
-  },
-  URL: {
-    isValid: (val) => {
-      try {
-        new URL(val);
-      } catch (_) {
-        return false;
-      }
-      return true;
-    },
-    msg: 'Please enter a valid url (e.g., https://www.google.com)'
-  },
-  EMAIL: {
-    isValid: isEmailAddress,
-    msg: 'Please enter a valid email address (e.g., person@example.com)'
-  }
+  REQUIRED: requiredValidator,
+  URL: urlValidator,
+  EMAIL: emailValidator,
 };
 
-//---------------------------------------------
-// Main Components
-//---------------------------------------------
-/*
-* Config options:
-* id, title, description
-* type (ENUM: 'text', 'multitext', 'select', 'sliding-checkbox', 'checkbox', 'radio')
-*  * type == 'text'
-*  * type == 'multitext'
-*  * type == 'select'
-*    * selectOptions: [{key: '', displayText: ''}] or ['', '']
-*    * isMulti: boolean - allow multiple simultaneous options
-*    * creatable: boolean - allow creating new values (not part of options)
-*  * type == 'radio'
-*    * options: [{name: '', displayText: '', type: ('boolean' || 'string')}]
-*       * if an option is of type string, a textbox will be added when selected.
-*  * type == 'checkbox'
-*    * toggleText
-*    * checkboxType
-*  * type === 'sliding-checkbox'
-* disabled: bool
-* placeholder, defaultValue,
-* style (for the formControl component)
-* inputStyle (for the input element)
-* onChange: func({key: '', value: '', isValid: boolean}),
-* validators: [{isValid: func, msg: ''}]
-* Accessibility: (defaults blank)
-*    * ariaDescribedby
-*    * ariaLevel
-*/
```

Comment:
also: deleted this because this information is more easily ascertained by looking at the configs. However, I would understand if we wished to keep this as well.